src/smolagents/models.py

```diff
@@ -1161,6 +1170,136 @@ def create_client(self):
+        custom_role_conversions (`dict[str, str]`, *optional*):
+            Custom role conversion mapping to convert message roles in others.
+            Useful for specific models that do not support specific message roles like "system".
+        flatten_messages_as_text (`bool`, default `False`):
+            Whether to flatten messages as text.
+
+    Example:
+        >>> bedrock_model = AmazonBedrockServerModel(
+        ...     model_id='us.amazon.nova-pro-v1:0',
+        ...     inference_config={'temperature': 0.7, 'max_tokens': 3000}
+        ... )
+    """
+
+    def __init__(
+        self,
+        model_id: str,
+        boto3_client=None,
+        inference_config: dict[str, Any] | None = None,
+        guardrail_config: dict[str, Any] | None = None,
+        bedrock_api_kwargs: dict[str, Any] | None = None,
+        custom_role_conversions: dict[str, str] | None = None,
+        flatten_messages_as_text: bool = False,
+        **kwargs,
+    ):
+        self.model_id = model_id
+        self.client = boto3_client or self.create_client()
+        self.inference_config = inference_config
+        self.guardrail_config = guardrail_config
+        self.bedrock_api_kwargs = bedrock_api_kwargs
+        self.custom_role_conversions = custom_role_conversions
+
+        super().__init__(
+            model_id=model_id,
+            custom_role_conversions=custom_role_conversions,
+            flatten_messages_as_text=flatten_messages_as_text,
+            **kwargs,
+        )
+
+    def create_client(self):
+        try:
+            import boto3
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                "Please install 'aws_sdk' extra to use AmazonBedrockServerModel: `pip install 'smolagents[aws_sdk]'`"
+            ) from e
+
+        return boto3.client("bedrock-runtime")
+
+    def __call__(
+        self,
+        messages: List[Dict[str, str]],
+        tools_to_call_from: Optional[List[Tool]] = None,
+        **kwargs,
+    ) -> ChatMessage:
+        # smolagents inject automatic parameters like end_stop, Observation and others
+        # The Bedrock API only supports the fields explicitly defined here.
+        # To ensure compliance, `kwargs` is set to empty, and customers must provide additional parameters using `bedrock_api_kwargs`.
+        kwargs = {}
+
+        # Bedrock only supports `assistant` and `user` roles.
+        # Many Bedrock models do not allow conversations to start with the `assistant` role, so the default is set to `user/user`.
+        # This parameter is retained for future model implementations and extended support.
+        if not self.custom_role_conversions:
+            self.custom_role_conversions = {
+                MessageRole.SYSTEM: MessageRole.USER,
+                MessageRole.ASSISTANT: MessageRole.USER,
+                MessageRole.TOOL_CALL: MessageRole.USER,
+                MessageRole.TOOL_RESPONSE: MessageRole.USER,
+            }
```

Comment:
This could be moved to `__init__` (see above).

Reply:
Agreed!